### PR TITLE
Add API docs to various forms of `performAndSave` function

### PR DESCRIPTION
**Pull request by crazytonyli**
Better late than never—I probably should have documented these APIs earlier.

## Regression Notes
1. Potential unintended areas of impact
None.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
None.

3. What automated tests I added (or what prevented me from doing so)
None.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/Utility/ContextManager.swift [hunk 1]
```diff
@@ -8,8 +8,25 @@ let ContextManagerModelNameCurrent = "$CURRENT"
 
 public protocol CoreDataStackSwift: CoreDataStack {
 
+    /// Execute the given block with a background context and save the changes _if the block does not throw an error_.
+    ///
+    /// This function _does not block_ its running thread. The block is executed in background and the its return value
```
Comment (mokagio) [inline WordPress/Classes/Utility/ContextManager.swift]:
```suggestion
    /// This function _does not block_ its running thread. The block is executed in background and its return value
```

File: WordPress/Classes/Utility/CoreDataStack.h [hunk 2]
```diff
@@ -9,8 +9,26 @@ NS_ASSUME_NONNULL_BEGIN
 - (void)saveContextAndWait:(NSManagedObjectContext *)context;
 - (void)saveContext:(NSManagedObjectContext *)context;
 - (void)saveContext:(NSManagedObjectContext *)context withCompletionBlock:(void (^ _Nullable)(void))completionBlock onQueue:(dispatch_queue_t)queue NS_SWIFT_NAME(save(_:completion:on:));
+
+/// Execute the given block with a background context and save the changes.
+///
+/// This function _blocks_ its running thread. The changed made by the `aBlock` argument are saved before this
+/// function returns.
+///
+/// - Parameter aBlock: A closure which uses the given `NSManagedObjectContext` to make Core Data model changes.
```
Comment (mokagio) [inline WordPress/Classes/Utility/CoreDataStack.h]: Nitpick for naming consistency.

```suggestion
/// - Parameter aBlock: A block which uses the given `NSManagedObjectContext` to make Core Data model changes.
```

File: WordPress/Classes/Utility/CoreDataStack.h [hunk 3]
```diff
@@ -9,8 +9,26 @@ NS_ASSUME_NONNULL_BEGIN
...
 - (void)performAndSaveUsingBlock:(void (^)(NSManagedObjectContext *context))aBlock;
+
+/// Execute the given block with a background context and save the changes _if the block does not throw an error_.
+///
+/// This function _does not block_ its running thread. The `aBlock` argument is executed in the background. The
+/// `completion` block is called after the Core Data model changes are saved.
+///
+/// - Parameters:
+///   - aBlock: A block which uses the given `NSManagedObjectContext` to make Core Data model changes.
+///   - completion: A closure which is called after the changes made by the `block` are saved.
```
Comment (mokagio) [inline WordPress/Classes/Utility/CoreDataStack.h]: Nitpick for naming consistency.

```suggestion
///   - completion: A block which is called after the changes made by the `block` are saved.
```

File: WordPress/Classes/Utility/ContextManager.swift [hunk 4]
```diff
@@ -8,8 +8,25 @@ let ContextManagerModelNameCurrent = "$CURRENT"
...
+    ///         are only saved if the block does not throw an error.
+    ///   - completion: A closure which is called with the `block`'s execution result, which is either an error thrown
+    ///         by the `block` or the return value of the `block`.
+    ///   - queue: A queue on which to execute the completion block.
     func performAndSave<T>(_ block: @escaping (NSManagedObjectContext) throws -> T, completion: ((Result<T, Error>) -> Void)?, on queue: DispatchQueue)
 
+    /// Execute the given block with a background context and save the changes _if the block does not throw an error_.
+    ///
+    /// - Parameter block: A closure that uses the given `NSManagedObjectContext` to make Core Data model changes.
+    /// The changes are only saved if the block does not throw an error.
```
Comment (mokagio) [inline WordPress/Classes/Utility/ContextManager.swift]: Nitpick: How do you feed about indenting the second line?

```suggestion
    /// - Parameter block: A closure that uses the given `NSManagedObjectContext` to make Core Data model changes.
    ///   The changes are only saved if the block does not throw an error.
```

No strong feelings. Xcode renders both versions as one would expect:

Screenshot 2023-03-08 at 2 34 32 pm

**Review by mokagio — APPROVED** [review]

File: WordPress/Classes/Utility/ContextManager.swift (hunk 4 above)
Comment (crazytonyli) [reply WordPress/Classes/Utility/ContextManager.swift]: Thanks! All of above are addressed in https://github.com/wordpress-mobile/WordPress-iOS/pull/20279/commits/a0b6b61fbc1f6af10b78aaa5d10c4d00da9d00dd

**Pull request merged**